ipc4: copier: Clarify the log and return value when the dai_copier is…

ujfalusi opened the pull request:
… not found

We are always returning 0 if the dai_copier is not found and never an error code. Make it clear by returning 0.

With that out of the way, it is clear that this is not an error, it is something which can happen.
Firmware log is a proof of that since it is flooded with <err> copier: comp:X 0xSOMEHEX failed to find dai comp or sink pipeline not running.

prints. Drop the log level to warn (but I think it could be even dbg

marc-hb commented:
Thanks Peter. This `<err>` has been spamming test logs with hundreds and even thousands of lines in long, daily test runs which makes it impossible to spot other, real errors.

- https://github.com/thesofproject/sof/issues/6734#issuecomment-1452731165

lyakh reviewed (commented):
Let me try to understand why that path is hit, hopefully today

lgirdwood commented on src/audio/copier/copier.c:
```diff
@@ -950,8 +950,8 @@ static int copier_comp_trigger(struct comp_dev *dev, int cmd)
-		comp_err(dev, "failed to find dai comp or sink pipeline not running.");
-		return ret;
+		comp_warn(dev, "failed to find dai comp or sink pipeline not running.");
+		return 0;
```
Ok, so this looks suspect without an inline comment. Can you add one.

ujfalusi replied:
@lyakh, would you add a patch to your #7210 to change the return value to some error code as  a last step? This should not happen now afaik, so if there is no copier found it is an error?
Then I can close this PR.

lgirdwood reviewed (commented):
Need a comment on wht we return 0 on a potential error.

marc-hb commented:
Superseded by #7210?

ujfalusi commented:
> Superseded by this?
> 
>     * [Fix "failed to find dai comp or sink pipeline not running." #7210](https://github.com/thesofproject/sof/pull/7210)

This PR is still valid with that PR as we would still return 0 and print with err level.

ujfalusi commented:
Changes since v1:
- drop to info log level
- reword the printed information and add additional comment for clarity
- with #7210 this is not going to happen
